src/System.Management.Automation/engine/serialization.cs

```diff
@@ -122,6 +122,45 @@ public static string Serialize(object source, int depth)
             return sb.ToString();
         }
 
+        /// <summary>
+        /// Serializes array of objects into PowerShell CliXml.
+        /// </summary>
+        /// <param name="source">The input objects to serialize.</param>
+        /// <param name="depth">The depth of the members to serialize.</param>
+        /// <param name="enumerate">Enumerates input objects and serializes one at a time.</param>
+        /// <returns>The serialized object, as CliXml.</returns>
+        internal static string Serialize(IEnumerable<object> source, int depth, bool enumerate)
+        {
+            StringBuilder sb = new();
+
+            XmlWriterSettings xmlSettings = new()
+            {
+                CloseOutput = true,
+                Encoding = Encoding.Unicode,
+                Indent = true,
+                OmitXmlDeclaration = true
+            };
+
+            XmlWriter xw = XmlWriter.Create(sb, xmlSettings);
+            Serializer serializer = new(xw, depth, useDepthFromTypes: true);
+
+            if (enumerate)
```

Comment:
How do this work if parameter is IEnumerable and enumerate is false?

Reply:
It will just serialize the entire enumerable at once. Do you think we should set a default for this? I was thinking for public API eventually you van configure this behaviour to enumerate or write entire enumerable at once.

Reply:
Basically if enumerate is false we get behaviour from the other public Serialize method.

Reply:
The question is, what if the argument is IEnumerable and enumerate = false? It doesn't seem to work.

Reply:
Interesting. Wonder if Serializer has issues with Ienumerable. Will need to write some Xunit tests to verify.

Reply:
@iSazonov I've changed to `IList<object> source`. Seems to have better behaviour. Added XUnit tests for both cases and it seems good, let me know what you think 🙂

Reply:
Formally LGTM.